src/BSON/Javascript.c

```diff
@@ -543,6 +529,10 @@ HashTable* php_phongo_javascript_get_properties_hash(zval* object, bool is_debug
 #endif
 
 	return props;
+
+failure:
+	FREE_HASHTABLE(props);
```

Comment:
I'm not sure this is correct. If you look at the definition of [`PHONGO_GET_PROPERTY_HASH_INIT_PROPS`](https://github.com/mongodb/mongo-php-driver/blob/master/php_phongo.h#L187), you'll see that we allocate _and_ initialize if we're returning properties for debugging purposes. Otherwise, we set `props` to the address of the intern's properties hash and conditionally alloc/init it if needed.

[`FREE_HASHTABLE`](http://lxr.jpauli.tech/source/xref/php-7.4/Zend/zend_alloc.h#236) corresponds to the initial allocation, but I think we're also missing a `zend_hash_destroy` to balance out the initialization (and clean up other fields that might have been added to the hash, such as the `code` string). If that makes sense, I'd propose we do so only for the case where `is_debug` is true.

For the non-debug case when we're working with the intern's properties hash, I think it'd be preferable to leave things as-is. On a failure, `scope` hasn't been assigned at all. It's possible that `code` has been set, but a future `get_properties` call could just update the field without a risk of leaking. That's what happens all the time when calling `get_properties` on our objects. We can trust that the hash table will ultimately get cleaned up during the `free_object` handler when the Javascript object dies.

Reply:
Very good point, I've adjusted the code and created a new `PHONGO_GET_PROPERTY_HASH_FREE_PROPS` macro that encapsulates the logic of checking `is_debug` and destroying the hash if so.